### PR TITLE
Feature/yonatanari/box component

### PR DESCRIPTION
**Pull request by 3dyonic**
https://monday.monday.com/boards/245345663/pulses/2623243775

#### Basic
- [ ] Used plop (`npm run plop`) to create a new component.
- [ ] PR has description.
- [ ] New component is functional and uses Hooks. 
- [ ] Component defines [`PropTypes`](https://reactjs.org/docs/typechecking-with-proptypes.html).
#### Style
- [ ] Styles are added to `NewComponent.modules.scss` file inside of the `NewComponent` folder.
- [ ] Component uses CSS Modules.
- [ ] Design is compatible with [Monday Design System](https://design.monday.com/).
#### Storybook
- [ ] Stories were added to `/src/NewComponent/__stories__/NewComponent.stories.js` file.
- [ ] Stories include all flows of using the component.
#### Tests
- [ ] Tests are compliant with [TESTING_README.md](TESTING_README.md) instructions.

### CODE REVIEW

File: package-lock.json
Comment (hadasfa) [inline package-lock.json:3090]: if u did not update the package.json file there should be not package-lock.json changes :)

File: src/components/Box/BoxConstants.js
Comment (hadasfa) [inline src/components/Box/BoxConstants.js:95]: in my personal opinion classes name as mx0 are less readable then marginxo
but this is only my opinion

File: src/components/Box/BoxCoreFunctions.scss
Comment (hadasfa) [inline src/components/Box/BoxCoreFunctions.scss:20]: I think this function is more general and maybe we can pust them in more general file which not only related to box?

Comment (hadasfa) [reply src/components/Box/BoxCoreFunctions.scss:20]: like general css functions file

File: src/components/Box/BoxCoreFunctions.scss
Comment (hadasfa) [inline src/components/Box/BoxCoreFunctions.scss:5]: if they should be imported from monday ui style let's remove it before merge the PR?

File: src/components/Box/BoxCoreMixin.scss
Comment (hadasfa) [inline src/components/Box/BoxCoreMixin.scss:2]: same comment about remove mock before merge

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:498]: maybe change api to be boolean prop: disabled will e more similar to other components? 
if we will have multiple opacity states in the feature it will be a scale and not with meaning (like spacing or border)

File: src/components/Box/__stories__/Box.stories.scss
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.scss:3]: not critical but why not css modules?

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (hadasfa) [inline src/components/Box/__tests__/box-snapshot-tests.jest.js:8]: didnt really get it. in our tests there is not elements inside the box... and in other implementation we will prefer they will replace our box with a mock (different developers tests should not break for each change we did in the box)

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (hadasfa) [inline src/components/Box/__tests__/box-snapshot-tests.jest.js:38]: and there is not tests :(

File: src/components/Box/__tests__/box-tests.jest.js
Comment (hadasfa) [inline src/components/Box/__tests__/box-tests.jest.js:6]: there is no need in behavior tests for this component

File: src/components/Box/Box.jsx
Comment (hadasfa) [inline src/components/Box/Box.jsx:2]: please fix all linter issues

File: src/components/Box/Box.jsx
Comment (hadasfa) [inline src/components/Box/Box.jsx:101]: bgcolor is different from the way we call background color in other components. please use the same naming :)

File: src/components/Box/Box.jsx
Comment (hadasfa) [inline src/components/Box/Box.jsx:100]: is color is font color? I know that the purpose was to use the same name as the css property but it a bit confusing... maybe something which indicate the prop purpose such as textColor?

File: src/components/Box/Box.jsx
Comment (hadasfa) [inline src/components/Box/Box.jsx:99]: in my opinion is not needed. if component is not visible the can just not render it. or use class name (it's not save any time using our variables)

File: src/components/Box/Box.jsx
Comment (orrgottlieb) [reply src/components/Box/Box.jsx:63]: either use left, top, right, bottom

or move to logical properties

File: src/components/Box/Box.jsx
Comment (orrgottlieb) [inline src/components/Box/Box.jsx:59]: for all overflow - remove visible as it is looks like a bug and we don't need to use it

File: src/components/Box/Box.jsx
Comment (orrgottlieb) [reply src/components/Box/Box.jsx:99]: +1

File: src/components/Box/BoxConstants.js
Comment (orrgottlieb) [inline src/components/Box/BoxConstants.js:4]: Nice!

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (orrgottlieb) [reply src/components/Box/__tests__/box-snapshot-tests.jest.js:38]: +1

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (orrgottlieb) [inline src/components/Box/__tests__/box-snapshot-tests.jest.js:35]: this component must have snapshot tests for each property

File: src/components/Box/__tests__/box-tests.jest.js
Comment (orrgottlieb) [reply src/components/Box/__tests__/box-tests.jest.js:6]: +1

File: src/components/Box/Box.jsx
Comment (orrgottlieb) [reply src/components/Box/Box.jsx:2]: +1

File: src/components/Box/BoxCoreFunctions.scss
Comment (3dyonic) [reply src/components/Box/BoxCoreFunctions.scss:20]: it should be a general function, its here as mock.

File: src/components/Box/BoxCoreMixin.scss
Comment (3dyonic) [reply src/components/Box/BoxCoreMixin.scss:2]: +1

File: src/components/Box/BoxConstants.js
Comment (3dyonic) [reply src/components/Box/BoxConstants.js:95]: do not agree. 
and anyway the style names are not visible to the api

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:101]: +1

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:99]: +1
for this phase i will remove these, but i do think it part of a good utility interface.
visibility can be removed when we use react :)

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:63]: +1
move to logical properties
will leave border. 
negative margin will be removed

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:75]: i do not agree, i think its very common.

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:95]: +1

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:498]: done.

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (3dyonic) [reply src/components/Box/__tests__/box-snapshot-tests.jest.js:38]: - unit tests complete: all props all values.
- snapshot test complete: all props first value.

File: src/components/Box/__tests__/box-tests.jest.js
Comment (3dyonic) [reply src/components/Box/__tests__/box-tests.jest.js:6]: removed not needed.

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:100]: changed to "text" prop

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:81]: removed

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:63]: overflow removed. will be added at later phase to allow box to use as scroll container. (TBD)

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:89]: i am leaving theme as we discussed, this is a common pattern.

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:95]: removed.

File: src/components/Box/BoxConstants.js
Comment (3dyonic) [reply src/components/Box/BoxConstants.js:4]: changed to boolean

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (3dyonic) [reply src/components/Box/__tests__/box-snapshot-tests.jest.js:35]: done. each prop got one with the first option defined. 
E.g per BackgroundColor i test only one option from the background colors that are available..

File: package-lock.json
Comment (3dyonic) [reply package-lock.json]: fixed.

File: src/components/Box/BoxConstants.js
Comment (3dyonic) [reply src/components/Box/BoxConstants.js:91]: per now kept my structure.

File: src/components/Box/__stories__/Box.stories.scss
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.scss]: changed.

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx]: done.

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:2]: @hadasfa i will require small assistant on this.

File: package.json
Comment (hadasfa) [inline package.json:129]: your package lock json file looks with to much changes. I think u run npm i without nvm use.

File: package.json
Comment (hadasfa) [inline package.json:135]: why?

File: src/components/Box/Box.module.scss
Comment (hadasfa) [inline src/components/Box/Box.module.scss:4]: can u explain to me what is this and what its purpose?

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:37]: text? why we need children and also text? the all idea is that the box can contain anything no?

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:51]: why using different scss convention for our convention :(

Comment (hadasfa) [reply src/components/Box/__stories__/Box.stories.mdx:51]: it will create confusion because people will see not consistent examples

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:88]: let's write guidelines? important things that people need to understand before using the component.
when to use it? common use case, main features (I think shadow, vibe spacing are important... what else?

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:94]: tip? maybe a case that you will prefer people to use different component instead of this? maybe dialog container can be confused with this component.

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [reply src/components/Box/__stories__/Box.stories.mdx:37]: ok this is text color... let's change the name to text color please it's confusing a bit

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:490]: let's add related components? Dialog container can be one, flex and responsive list are also options

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx:42]: I think it should be title="Layout/Box"

Comment (hadasfa) [reply src/components/Box/__stories__/Box.stories.mdx:42]: but not sure

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (hadasfa) [inline src/components/Box/__tests__/box-snapshot-tests.jest.js:6]: this comment is for the developer writing the test, let's remove it before merge

File: src/components/Box/__tests__/box-tests.jest.js
Comment (hadasfa) [reply src/components/Box/__tests__/box-tests.jest.js:5]: but  there are still tests here? the snapshot check existence of class name

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:88]: i will push this through with Evgeny.

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:51]: this is consistent with our storybook classes.
let's discuss it, have not problem to change it.

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:37]: ok

File: src/components/Box/Box.module.scss
Comment (3dyonic) [reply src/components/Box/Box.module.scss:4]: loading dependencies from "monday-ui-styles"
all the magic come from it.
we load the mixin and its supporting functions.
since there is no effect in bundle size we can load it all instead of declaring the parts.instead of loading 8 differant files or declaring differnat file in any component we just load the common when we want the functions,mixins,variables from monday-ui-styles.

File: package.json
Comment (3dyonic) [reply package.json:135]: Orr did it so jest can detect the classNames without the css modules.

File: package.json
Comment (3dyonic) [reply package.json:129]: i reverted to masters, then uses nvm use, then did npm i , this is the result. :(

Comment (3dyonic) [reply package.json:129]: maybe due to npm modification?

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:94]: will consult with Evgeny

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:490]: ok

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx]: typo, corrected

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx:490]: They are missing from the related component descriptions.
let sync on that in our meeting today.

File: src/components/Box/__stories__/Box.stories.mdx
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx]: done

File: package.json
Comment (orrgottlieb) [inline package.json:135]: @3dyonic what is this ?

File: src/components/Box/BoxConstants.js
Comment (orrgottlieb) [inline src/components/Box/BoxConstants.js:7]: for all borders - either boolean or add layout-border-color

File: src/components/Box/BoxConstants.js
Comment (orrgottlieb) [inline src/components/Box/BoxConstants.js:252]: for the following colors: hoverColor, selectedColor, iconColor,LinkColor, placeholderColor - IMO they shouldn't be an option to use them in a container - they are either too component specific or they are shouldn't be used

File: src/components/Box/Box.jsx
Comment (orrgottlieb) [inline src/components/Box/Box.jsx:80]: please align with Flex and or Clickable

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/components/Box/Box.jsx
Comment (hadasfa) [reply src/components/Box/Box.jsx]: agree

File: package.json
Comment (hadasfa) [reply package.json:135]: +1

File: src/components/Box/BoxConstants.js
Comment (hadasfa) [reply src/components/Box/BoxConstants.js:252]: +1

File: src/components/Box/Box.jsx
Comment (hadasfa) [reply src/components/Box/Box.jsx:80]: I am not sure flex is the right choice here, it is a bit to opinionated (I don't think we should decide for the developer what would be the why to spread the container children).
I did not see on click prop, but I agree that if we do support on click we need to use Clickable component for that.

File: src/components/Box/Box.jsx
Comment (hadasfa) [inline src/components/Box/Box.jsx:99]: what @3dyonic and @orrgottlieb  are you decide regarding marginx, marginy, paddingx, paddingy in the end?

File: src/components/Box/Box.jsx
Comment (hadasfa) [reply src/components/Box/Box.jsx:80]: Maybe passing here element type like we did in the clickable component (@3dyonic you can see the clickable component for reference)

File: src/components/Box/__stories__/Box.stories.module.scss
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.module.scss:3]: it's not critical, but for next time: the monday prefix is not required on css modules class names :)

File: src/components/Box/__tests__/box-snapshot-tests.jest.js
Comment (hadasfa) [inline src/components/Box/__tests__/box-snapshot-tests.jest.js:144]: Super important: let's add a test for "with children prop."
Without children prop support this component will be broken.

File: src/storybook/components/related-components/component-description-map.js
Comment (hadasfa) [inline src/storybook/components/related-components/component-description-map.js:129]: yay! more descriptions  🥇

File: src/styles/utilities/_utilities-settings.scss
Comment (hadasfa) [inline src/styles/utilities/_utilities-settings.scss:3]: We already merge the monday ui styles pr. I think we can delete the mock now.

**Review by hadasfa — CHANGES_REQUESTED** [review]

File: src/styles/utilities/_utilities-settings.scss
Comment (hadasfa) [reply src/styles/utilities/_utilities-settings.scss:3]: Also please pull from master before merge (you have some conflicts there :) )

Comment (3dyonic) [reply src/styles/utilities/_utilities-settings.scss:3]: this is not the mock :) 
This is the settings that uses the mixin from styles.

File: src/components/Box/Box.jsx
Comment (3dyonic) [reply src/components/Box/Box.jsx:99]: there are in.

File: src/components/Box/__stories__/Box.stories.module.scss
Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.module.scss]: done

File: package.json
Comment (hadasfa) [reply package.json:135]: don't forget to push the delete of this before merge to master

**Review by hadasfa — APPROVED** [review]

File: src/components/Box/Box.jsx
Comment (hadasfa) [inline src/components/Box/Box.jsx]: why we removed none? what if we do not want any margin?

File: src/components/Box/BoxConstants.js
Comment (hadasfa) [reply src/components/Box/BoxConstants.js:7]: is this note still relevant? @orrgottlieb @3dyonic ?

File: src/components/Box/BoxConstants.js
Comment (hadasfa) [inline src/components/Box/BoxConstants.js:17]: why we do not support none anymore?

File: src/components/Box/__stories__/Box.stories.mdx
Comment (hadasfa) [inline src/components/Box/__stories__/Box.stories.mdx]: in general we do like add use cases and examples from the real world (monday products) under use case and examples section. but we can do this in a different iteration under different task.

Comment (3dyonic) [reply src/components/Box/__stories__/Box.stories.mdx]: @meiravron is working on that.
we has 2 examples but she wanted to rethink it.

**Pull request merged**
